include/cista/exception.h

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+namespace cista {
+
+template <typename _E>
+void throw_exception(_E&& e) {
```

Comment:
```suggestion
template <typename E>
void throw_exception(E&& e) {
```

Any reason to prefix with `_`?